app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt

```diff
@@ -106,6 +111,9 @@ class DefaultSessionControlControllerTest {
 
     @Before
     fun setup() {
+
+        mockkObject(NavGraphProvider)
```

Comment:
I see: we need to duplicate this high level logic ("stub the blocking method") because this isn't a Robolectric test so it isn't handled globally.

However, I think we can do better with the implementation. I'm concerned with:
- duplicating lines of code in many tests that can change in the future
- making mistakes in duplication (e.g. we didn't call `unmockkObject` here)
- devs getting an error they can't interpret when they call `navigateBlockingForAsyncNavGraph` in tests (when the solution is to add these two mysterious lines of code)

I think I have an idea: since it's discouraged to modify a global "runner"/"application"/"superclass test class" in JUnit (like we did with robolectric), let's use JUnit Rules – that will mostly solve problems 1 & 2. And when we throw the exception, we can add a line to it, "If this exception is thrown during testing, please add the `WhateverWeCallItRule`", which solves problem 3, the discoverability issue.

What do you think?

I only skimmed it but this looks like a good article on Rules:  https://ncorti.com/blog/junit-rules